Author: MSP-Greg <Greg.mpls@gmail.com>
Date:   Sat Sep 26 12:40:17 2020 -0500

    Windows: fix 'Use Tempfile.create instead of Tempfile.open in test_io.rb'

    Windows will not allow a file to be unlinked if any file handles exist

diff --git a/test/ruby/test_io.rb b/test/ruby/test_io.rb
index 4bb8a1d1ff..ee2336bfa7 100644
--- a/test/ruby/test_io.rb
+++ b/test/ruby/test_io.rb
@@ -2741,15 +2741,17 @@ def test_flush_in_finalizer1
       path = t.path
       t.close
       fds = []
+      ios = []
       assert_nothing_raised(TypeError, bug3910) do
         500.times {
           f = File.open(path, "w")
           f.instance_variable_set(:@test_flush_in_finalizer1, true)
           fds << f.fileno
+          ios << f
           f.print "hoge"
         }
       end
-      t
+      ios.each(&:close) if windows?
     }
   ensure
     ObjectSpace.each_object(File) {|f|
